### DISCUSSION TITLE
How to use GridPatchDataset generate non-overlap patch data?

### DISCUSSION THREAD

**Discussion by lanslotttTT** [Project-MONAI/MONAI#4058] [Q&A] [answered]
I want to use Monai produce **non-overlap** patch data with GridPatchDataset,it seemd that  this class not support dict dataset,so I got the error, like below 
```python
    patch_size_ = get_valid_patch_size(arr.shape, patch_size)
AttributeError: 'dict' object has no attribute 'shape'
```
what should I do?

**Comment by Nic-Ma** [comment, marked as answer]
Hi @lanslotttTT ,

Thanks for your feedback.
Looks we should add support for `dict` data, like a `PatchIterd` class.
I will take a deeper look, track in https://github.com/Project-MONAI/MONAI/issues/4060.

Thanks.

**Reply by Nic-Ma** [reply]
Submitted PR for it: https://github.com/Project-MONAI/MONAI/pull/4061.

Thanks.
